prov/sockets/src/sock_fabric.c

```diff
@@ -580,20 +581,21 @@ SOCKETS_INI
 {
 	char *value;
 
-	if ((value = getenv("SOCK_PE_WAITTIME"))) {
+	fi_var_register(&sock_prov, "pe_waittime",
+                        "How many miliseconds to spin while waiting for progress");
+	fi_var_get(&sock_prov, "pe_waittime", &value);
+	if (value)
 		sock_pe_waittime = atoi(value);
-		SOCK_LOG_INFO("SOCK_PE_WAITTIME = %d\n", sock_pe_waittime);
-	}
-	
+
 	fastlock_init(&sock_list_lock);
 	dlist_init(&sock_fab_list);
 	dlist_init(&sock_dom_list);
 #if ENABLE_DEBUG
-	if(getenv("SOCK_DGRAM_DROP_RATE") != NULL) {
-		sock_dgram_drop_rate =  strtol(getenv("SOCK_DGRAM_DROP_RATE"), NULL, 10);
-		if(sock_dgram_drop_rate)
-			SOCK_LOG_INFO("SOCK_DGRAM_DROP_RATE = %d\n", sock_dgram_drop_rate);
-	}
+	fi_var_register(&sock_prov, "dgram_drop_rate",
+			"Drop every Nth dgram frame (debug only)");
+	fi_var_get(&sock_prov, "dgram_drop_rate", &value);
+	if (value)
+		sock_dgram_drop_rate = strtol(value, NULL, 10);
```

Comment:
@jithinjosepkl - This is separate from this patch, but can you look at moving reading these values until immediately before they are needed?